YahooFantasy.mjs

```diff
@@ -48,34 +48,51 @@ class YahooFantasy {
   }
 
   api(...args) {
-    const method = args.shift();
-    const url = args.shift();
-    const cb = args.pop();
-    let postData = false;
+    var options = {};
+    let cb = () => {};
+    let responseMapper = x => x;
```

Comment:
Every method that called this passed a callback that mapped the response and invoked their caller's callback.

So instead of changing each method to support a promise, I just let them inform the central API method of the mapping function and let the central function deal with promises/callbacks.

Adds to the API, but results in less work per individual resource.

Reply:
hmmm this certainly seems simpler... sorry again, so busy at work right now, and that pays the bills so it takes priority. Let me take a deeper look over the weekend and see which one I like more.

Reply:
I think I agree about this being simpler. 

I definitely agree about real bills taking priority,

No worries, no rush.

Reply:
Can you just explain the responseMapper? I don't think I totally understood it the first time I saw it, and now I'm even more confused... I think...

Reply:
So, the general pattern I saw for most functions in this library is:

1. Convert some input to a URI
2. Invoke `yf.api` with that URI
3. Convert the result from that call into an easier-to-use object (drop fantasy_content for instance)

responseMapper is step 3. By passing it to `yf.api`, `yf.api` can directly invoke it and then the caller's callback, lessening the load on each function of converting the response to a promise.

I had hoped the `request` library might have provided some such mapping function, but the closest [it offers] is a [jsonReviver function parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse), but that transforms each key/value in the response body object to another, meaning that I wouldn't be able to map the root fantasy_content object to multiple root objects.

Reply:
It is pretty much the `transformResponse` parameter [axios offers](https://github.com/axios/axios#request-config)

I've only ever used `request` (and `request-promise`), but `axios` is intriguing.